src/app/pages/wallets/wallet-details/pages-wallet-details.component.ts

```diff
@@ -0,0 +1,129 @@
+})
+export class PagesWalletDetailsComponent implements OnInit, OnDestroy {
+
+  public selectTransactionsTypes: Record<string, WalletTransactionType | ''> = {
+    'All transactions': '',
+    'Incomes': WalletTransactionType.Incomes,
+    'Expenses': WalletTransactionType.Expenses,
+  }
+
+  public selectedTransactionType: Record<string, WalletTransactionType | ''> = this.selectTransactionsTypes;
+
+  public transactionsTypeForm = new FormControl<WalletTransactionType | ''>('');
+
+  private transactionTypeSub?: Subscription;
+
+  public walletTransactionType: typeof WalletTransactionType = WalletTransactionType;
+
+  public displayedColumns: string[] = ['date', 'description', 'amount', 'actions'];
+
+  public walletsManagementItem?: WalletsManagementItem;
+
+  public displayedTransactions: WalletsDetailsTransaction[] = [];
+
+  public walletsDetailsData: TDataState<WalletsDetailsTransaction[]> = {
+    data: null,
+    hasError: false,
+    isLoading: true,
+  }
+
+  public constructor(
+    private readonly pagesWalletDetailsService: PagesWalletDetailsService,
+    private readonly myWalletsService: PagesWalletsManagementService,
+    private readonly systemNotificationsService: SystemNotificationsService,
+    private readonly dialog: MatDialog,
+    private readonly activatedRoute: ActivatedRoute,
+  ) { }
+
+  public ngOnInit(): void {
+    this.activatedRoute.data.subscribe( ({ wallet }) => {
+      if (!(wallet instanceof WalletsManagementItem)) {
+        return;
+      }
+
+      this.walletsManagementItem = wallet;
+      this.initWalletDetails(wallet);
+    });
+
+    this.transactionTypeSub = this.transactionsTypeForm.valueChanges.subscribe((data: WalletTransactionType | '' | null) => {
+      this.filterTransactions(data ?? '');
+    });
+  }
+
+  private initWalletDetails(wallet: WalletsManagementItem): void {
+    this.pagesWalletDetailsService.getWalletTransactions(wallet.id!).subscribe({
+      next: (data) => {
+        this.displayedTransactions = data;
+        this.walletsDetailsData = {
+          data,
+          hasError: false,
+          isLoading: false,
+        }
+      },
+      error: () => {
+        this.walletsDetailsData = {
+          data: null,
+          isLoading: false,
+          hasError: true,
+        };
+      },
+    })
+  }
+
+  private filterTransactions(type: WalletTransactionType | ''): void {
+    this.displayedTransactions = this.walletsDetailsData.data!.filter( (item: WalletsDetailsTransaction) =>  (type === '') ? true : type.toString() === item.type.toString() );
+  }
+
+  private openWalletModal(wallet?: WalletsManagementItem): Observable<IWalletModalData | undefined> {
+    const dialogRef = this.dialog.open<PagesWalletsManagementEditorComponent, WalletsManagementItem | undefined, IWalletModalData>(PagesWalletsManagementEditorComponent, {
+      data: wallet,
+    });
+
+    return dialogRef.afterClosed();
+  }
+
+  public handleWalletEdit(): void {
```

Comment:
Looks like `openWalletModal`, `handleWalletEdit` and `updateWallet` were copied from wallets management component. In that to reduce duplication of code I would consider to move all related code from both components to a service that will be reused in both components, it can be even 1 method `openWalletEditor` that:
- will open modal for creating/editing wallet
- will make a proper request to the api
- will show success/failure messages
- will return created/edited wallet in the end or null (or even throw error) if whole operation failed for some reason